### PR TITLE
Checking label lists for labels

### PR DESCRIPTION
**Pull request by yullin-ms**
Fixes OS: 18576812
In defer parsing, when continue or break to label is in the same statement as label; check current label lists for labels.

### CODE REVIEW

File: test/Basics/Labels.js [hunk 1]
```diff
@@ -338,6 +338,21 @@ testNoSyntaxErrorWithDanglingStaticAfterLabel=
 testRuntimeErrorWithDanglingStaticAfterLabel=
     `L: static
     x = 0`
+
+testSyntaxErrorWithContinueAfterLabel=
+    `function fn() {
+        L:
+            continue L;
+    }
+    fn1()`
```
Comment (boingoing) [inline test/Basics/Labels.js]: Is the SyntaxError we're looking for an early syntax error? If so, we shouldn't have to call any function to hit it.

File: test/Basics/Labels.js [hunk 2]
```diff
@@ -338,6 +338,21 @@ testNoSyntaxErrorWithDanglingStaticAfterLabel=
...
     x = 0`
+
+testSyntaxErrorWithContinueAfterLabel=
+    `function fn() {
+        L:
+            continue L;
+    }
+    fn1()`
+
+testReferenceErrorWithBreakAfterLabel=
```
Comment (boingoing) [inline test/Basics/Labels.js]: Does it make sense to make this into a test case which doesn't throw? Use `assert.doesNotThrow` below? The reference error we're testing for here has nothing to do with the label binding.

File: test/Basics/Labels.js (hunk 1 above)
Comment (yullin-ms) [reply test/Basics/Labels.js]: Both have been corrected.

File: lib/Parser/Parse.cpp [hunk 3]
```diff
@@ -10509,12 +10509,24 @@ ParseNodePtr Parser::ParseStatement()
             else
             {
                 this->GetScanner()->Scan();
+                // Check if label is found within the current label id list.
+                for (LabelId* pLabelId = pLabelIdList; pLabelId; pLabelId = pLabelId->next)
```
[the remark this answers is not in the document]
Comment (yullin-ms) [reply lib/Parser/Parse.cpp]: That have been updated.

**Pull request merged**